app/controllers/concerns/shopify_app/ensure_installed.rb

```diff
@@ -58,5 +63,12 @@ def shop_login
 
       url.to_s
     end
+
+    def validate_emebedded_session_is_active
+      client = ShopifyAPI::Clients::Rest::Admin.new(session: installed_shop_session)
+      client.get(path: "shop")
+    rescue ShopifyAPI::Errors::HttpResponseError => error
+      redirect_for_embedded if error.code == 401 # unauthorized due to uninstall
```

Comment:
Would this be a `redirected_for_embedded` situation? I think we would call this when NOT embedded, right?

Reply:
ah ya, that's not clear because the conditional for this method is listed in the `before_action` declaration. I'll fix

Reply:
As I understand it, we should only be doing this session sanity check if it is embedded, right?

Reply:
We should redirect to login, which we sadly call `login_url` for reasons 🤦